eip_7594/src/lib.rs

```diff
@@ -343,6 +372,64 @@ fn kzg_commitment_inclusion_proof<P: Preset>(
     proof
 }
 
+/**
+ * Return the sample count if allowing failures.
+ *
+ * This helper demonstrates how to calculate the number of columns to query per slot when
+ * allowing given number of failures, assuming uniform random selection without replacement.
+*/
+pub fn get_extended_sample_count(allowed_failures: u64) -> u64 {
+    // check that `allowed_failures` within the accepted range [0 -> NUMBER_OF_COLUMNS // 2]
+    assert!((0..(NumberOfColumns::U64 / 2)).contains(&allowed_failures));
+
+    // missing chunks for more than a half is the worst case
+    let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
+
+    // the probability of successful sampling of an unavailable block
+    let false_positive_threshold = hypergeom_cdf(
+        0,
+        NumberOfColumns::U64,
+        worst_case_missing,
+        SAMPLES_PER_SLOT,
+    );
+
+    // number of unique column IDs
+    let mut sample_count = SAMPLES_PER_SLOT;
+    while sample_count <= NumberOfColumns::U64 + 1 {
+        // TODO(feature/das): change variable name `x` to a suitable one
+        let x = hypergeom_cdf(
+            allowed_failures,
+            NumberOfColumns::U64,
+            worst_case_missing,
+            sample_count,
+        );
+        if x <= false_positive_threshold {
+            break;
+        }
+        sample_count += 1;
+    }
+    return sample_count;
+}
+
+fn hypergeom_cdf(k: u64, m: u64, n: u64, big_n: u64) -> f64 {
+    let mut sum = 0_f64;
+    for i in 0..(k + 1) {
+        sum += math_comb(n, i) * math_comb(m - n, big_n - i) / math_comb(m, big_n);
+    }
+    sum
+}
```

Comment:
minor, but can use `..=`

```rust
fn hypergeom_cdf(k: u64, m: u64, n: u64, big_n: u64) -> f64 {
    let mut sum = 0_f64;
    for i in 0..=k {
        sum += math_comb(n, i) * math_comb(m - n, big_n - i) / math_comb(m, big_n);
    }
    sum
}
```

Reply:
we can actually do better; handle edge cases and compute denominator only once

```rust
fn hypergeom_cdf(k: u64, m: u64, n: u64, big_n: u64) -> f64 {
    if big_n > m || n > m { // Invalid input
        return 0.0;
    }
    let denominator = math_comb(m, big_n);
    let mut sum = 0.0;
    for i in 0..=k {
        if i <= n && (big_n - i) <= (m - n) {
            sum += math_comb(n, i) * math_comb(m - n, big_n - i) / denominator;
        }
    }
    sum
}

Reply:
you did calculated well, I haven’t thought about that. I just try to comply the [specs](https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/das-core.md#get_extended_sample_count), and improve the performance over time:

Reply:
I guess you could make a PR to update specs if you found that thing could be improved